Fix xenia-core build on macOS

conradev opened the pull request:
I had to make a few different changes to get it to compile:

- I force Xcode to only build for x86_64, even on ARM Macs
- I switched from OSAtomic, which was deprecated and now gone, to clang's builtins (which are compatible with GCC's)
- I updated the `renderdoc` API header to the latest version and it fixed the build, but that probably broke something else
- Fixed some weirdness with `uint64_t` and `size_t` conversion.

`disruptorplus` also needs a small type clarification in order to compile on macOS, but `xenia` doesn't have a fork of it yet

conradev commented on src/xenia/base/utf8.cc:
```diff
@@ -241,7 +241,6 @@ std::string_view::size_type find_any_of(const std::string_view haystack,
 
   auto [haystack_begin, haystack_end] = make_citer(haystack);
   auto [needle_begin, needle_end] = make_citer(needles);
-  auto needle_count = count(needles);
```
These were unused variables

Triang3l commented on premake5.lua:
```diff
@@ -17,7 +17,7 @@ else
 end
 
-includedirs({
+sysincludedirs({
```
`third_party` may count as a system include directory, but why should it be changed for `src` and for the root?

Triang3l commented on premake5.lua:
```diff
@@ -211,6 +211,12 @@ workspace("xenia")
+    elseif os.istarget("macosx") then
+      platforms({"Mac"})
+      xcodebuildsettings({           
+        ["GCC_WARN_UNUSED_VARIABLE"] = "NO",
```
Is this an issue with something like unused arguments of virtual function overrides primarily (or preprocessor conditionals or assertions), or do we have a large number of truly unused variables?

JoelLinn replied:
Yes we have those, i believe they crept in because the warning was disabled for a number of valid cases…

Triang3l commented on src/xenia/base/atomic.h:
```diff
@@ -21,41 +21,7 @@ namespace xe {
 // Original link (dead):
 // https://developer.apple.com/library/mac/#documentation/DriversKernelHardware/Reference/libkern_ref/OSAtomic_h/
```
These four comment lines are now outdated.

Triang3l commented on xenia-build:
```diff
@@ -1690,6 +1697,9 @@ class DevenvCommand(Command):
                 return 1
             print('Launching Visual Studio...')
+        elif sys.platform == 'darwin':
+            print('Launching Xcpde...')
```
This part is a duplicate of #1895 (there is a comment about the typo in the discussion of that pull request) — they should be separate, or this pull request should be superseding the xenia-build one.

JoelLinn commented on src/xenia/base/math.h:
```diff
@@ -139,6 +139,7 @@ static_assert(sizeof(unsigned int) == sizeof(uint32_t));
 static_assert(sizeof(unsigned long long) == sizeof(uint64_t));
 inline uint32_t bit_count(uint32_t v) { return __builtin_popcount(v); }
 inline uint32_t bit_count(uint64_t v) { return __builtin_popcountll(v); }
+inline uint32_t bit_count(size_t v) { return __builtin_popcountl(v); }
```
Iirc when i was trying to compile on mac, it was only one case in the arena alloc

Triang3l replied:
That call can be changed to `xe::is_pow2`.

Triang3l replied:
That call can be changed to `xe::is_pow2`, though it returns true for 0 as well — it can be handled with an additional assertion or with `align = std::max(align, size_t(1))`.

The overall handling of the alignment of `Chunk` there is pretty fragile, by the way — assuming that `malloc` returns 16-byte-aligned memory, while it's true only on 64-bit x86 systems; on 32-bit x86, `malloc` only guarantees 8-byte alignment, so it's better to use an aligned malloc function provided by the platform explicitly. Also that assertion says that the number 16 is the alignment of `Chunk`, which suggests that `alignof` may be used there instead. However, `Chunk` requires the alignment of a pointer actually — and there's no `alignas` to overalign the type. But there's no clear requirement that the `Chunk` class itself in isolation must be aligned. However, it may still be nice to move that 16 to a `static constexpr` with a static assertion that it must be `>= alignof(Chunk)`.

conradev commented:
Updated to address feedback, and backed out the `renderdoc` changes (as they aren't needed for `xenia-core`). I also re-enabled the unused variables warning and fixed the unused variables
